File: DuktapeEngineDemo/src/com/furture/react/activity/test/TestRef.java
package com.furture.react.activity.test;

import android.util.Log;

import com.furture.react.DLog;
import com.furture.react.JSRef;

public class TestRef {
	JSRef jsRef;
	public TestRef(JSRef jsRef){
		this.jsRef = jsRef;
	}
	
	public Object callJsRef(JSRef args){
		if(jsRef != args){
			throw new RuntimeException("TestRef Failed Must Same ref");
		}
		DLog.e("ScriptEngine", "ScriptEngine TestRef " + args.getRef());
		return jsRef;
	}
}

File: DuktapeCore/duktapetest/src/main/assets/testRef.js
importClass("com.furture.react.activity.test.TestRef")


/**
 *  测试引用，
 * javascript 到java的引用，可以是多个javascript对象引用一个Java对象。保持单一引用并不好。
 * java引用同一个javascript对象是单一的。
 */

var a = {say:function(){
	  print("save hello world");
}};

var testRef = new TestRef(a);

var b = testRef.callJsRef(a);

for(var i=0; i<10; i++){
	testRef.callJsRef(a);
	testRef.callJsRef(b);
}

b.say();
a.say();
print(a == b)

for(var i=0; i< 10; i++){
     testRef.javaRef();
}


true;